8348625: [21u, 17u] Revert JDK-8185862 to restore old java.awt.headless behavior on Windows

RealCLanger opened the pull request:
This basically reverts 78e35ea113bed4a5f66225836c887a45696bbf44 to restore old behavior with regards to headless handling on Windows.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8348625](https://bugs.openjdk.org/browse/JDK-8348625) needs maintainer approval

### Issue
 * [JDK-8348625](https://bugs.openjdk.org/browse/JDK-8348625): [21u, 17u] Revert JDK-8185862 to restore old java.awt.headless behavior on Windows (**Bug** - P2 - Approved)


### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk21u-dev.git pull/1359/head:pull/1359` \
`$ git checkout pull/1359`

Update a local copy of the PR: \
`$ git checkout pull/1359` \
`$ git pull https://git.openjdk.org/jdk21u-dev.git pull/1359/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1359`

View PR using the GUI difftool: \
`$ git pr show -t 1359`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk21u-dev/pull/1359.diff">https://git.openjdk.org/jdk21u-dev/pull/1359.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk21u-dev/pull/1359#issuecomment-2614402838)
</details>

jerboaa reviewed (approved):
Differences to a revert of 78e35ea113bed4a5f66225836c887a45696bbf44 are:

- `src/java.desktop/share/classes/java/awt/GraphicsEnvironment.java` unused import isn't back re-introduced. OK
- `src/java.desktop/windows/classes/sun/awt/Win32GraphicsEnvironment.java` extra space in comment not back re-introduced. OK.
- `src/java.desktop/windows/native/libawt/windows/awt_Win32GraphicsDevice.cpp`: `VERIFY` statements kept. Basically an assert, which should be fine.
- `src/java.desktop/windows/native/libawt/windows/awt_Win32GraphicsEnv.cpp` the `return` in the failure case is not re-introduced. We are done with the method anyway. OK.

Seems fine to me. But it would be good if a windows person could take a look and test it as well. @fthevenet perhaps?

fthevenet commented:
@jerboaa I'm going to look into reverting the changes made to our internal test infra following the introduction of the original patch in 17.0.12 and verify the tests that failed at the time now pass with this PR.

jerboaa commented:
> @jerboaa I'm going to look into reverting the changes made to our internal test infra following the introduction of the original patch in 17.0.12 and verify the tests that failed at the time now pass with this PR.

Thanks. Since this is a 21u-dev PR it should be tested by the tests prior to the 21.0.4 release (which introduced this regression).

fthevenet commented:
Of course, you're right. What I really meant to say was introduced in the July 24 update (21.0.4 and 17.0.12).

RealCLanger commented:
I fixed the title and I added an exclusion of the test jdk/java/awt/font/JNICheck/FreeTypeScalerJNICheck.java for Windows debug builds because these could bring up the assertions in headless test environments.

This kind:
*********************
AWT Assertion Failure
*********************
hBM != 0
File '...\\openjdk-21u-dev-windows_x86_64-dbg\\jdk\\src\\java.desktop\\windows\\native\\libawt\\windows\\awt_Win32GraphicsDevice.cpp', at line 185
GetLastError() is 6 : The handle is invalid.

Do you want to break into the debugger?
*********************

jerboaa reviewed (approved):
OK.

fthevenet commented:
I figured a quick and easy way to reproduce the issue locally, by using this [srvany clone](https://github.com/birkett/srvany-ng) to run a sample swing application as a Windows service.
When running this with a 21.0.3 JDK, the application runs but fails in 21.0.6 with a `java.awt.HeadlessException`, as expected.
When attempting the same thing with an image that includes this PR, the application runs again.

jerboaa commented:
> When attempting the same thing with an image that includes this PR, the application runs again.

Thanks for testing it, @fthevenet!
